### PR TITLE
JCLOUDS-1133: Use latest Java package available

### PR DESCRIPTION
**Pull request by neykov**
Xenial doesn't have Java 6 package.
Select the java package dynamically based on what's available. Don't hardcode Java 6.

### CODE REVIEW

**Comment by nacx** [comment]
FTR: https://issues.apache.org/jira/browse/JCLOUDS-1133

Thanks @neykov! For consistency, could you apply the same pattern to the packages that are installed with yum?

File: compute/src/test/resources/initscript_with_java.sh [hunk 1]
```diff
@@ -177,7 +177,11 @@ function installOpenJDK() {
     echo reusing JAVA_HOME $JAVA_HOME
   else
     if which dpkg &> /dev/null; then
-      apt-get-update && apt-get-install openjdk-6-jdk
+      apt-get update && \
+          PACKAGE=`apt-cache search --names-only '^openjdk-.-jdk$' | sort -r | cut -d' ' -f1 | head -1` && \
+          [ ! -z "$PACKAGE" ] && \
+          DEBIAN_FRONTEND=noninteractive && \
+          {  apt-get -q -y install $PACKAGE-headless || apt-get -q -y install $PACKAGE; }
```
Comment (nacx) [inline compute/src/test/resources/initscript_with_java.sh]: Better use the [apt-get-install](https://github.com/jclouds/jclouds/blob/master/scriptbuilder/src/main/resources/functions/setupPublicCurl.sh#L18) alias.

Comment (neykov) [inline compute/src/test/resources/initscript_with_java.sh]: Oh it's an alias, wondered how it was working before :). Will update.

**Comment by neykov** [comment]
Updated PR to use `apt-get-install` and support for yum.

**Comment by nacx** [comment]
Pushed to [master](http://git-wip-us.apache.org/repos/asf/jclouds/commit/325a6890) and [2.0.x](http://git-wip-us.apache.org/repos/asf/jclouds/commit/86f518dd). Apologies for the delay @neykov!

**Pull request closed without merging**